vllm/v1/core/sched/scheduler.py

```diff
@@ -411,7 +417,8 @@ def schedule(self) -> SchedulerOutput:
                 # KVConnector: update internal state after allocation.
                 # This information is used to determine if a load is
                 # needed for this request.
-                if self.connector is not None:
+                if num_external_computed_tokens:
```

Comment:
I don't think this is right. We should let the connector decide what to if `num_external_computed_tokens=0`

For instance, this will cause a memory leak on the P worker if the D worker has a full prefix cache hit.

Reply:
Thanks @robertgshaw2-redhat I considered this and thought it was ok but got P/D mixed up and you're right.

It will mean we need to rethink some things w.r.t. the multi-connector impl though since we currently cycle through the connectors in `get_num_new_matched_tokens` until the first one that returns nonzero.

I think this could be addressed by handling that case for nixl in `get_num_new_matched_tokens` itself, I'll add a change for that.